src/GitHubExtension/DeveloperId/DeveloperIdProvider.cs

```diff
@@ -296,14 +313,20 @@ private DeveloperId CreateOrUpdateDeveloperIdFromOauthRequest(OAuthRequest oauth
         return newDeveloperId;
     }
 
-    private void RestoreDeveloperIds(IEnumerable<string> loginIds)
+    private void RestoreDeveloperIds(IEnumerable<string> loginIdsAndUrls)

```

Comment:
A bit fishy.  Having a string that could either be loginIds or URLs.    How would a user get into a state where `loginIdsAndUrls` can contain login ids and urls?

Reply:
I'll leave a comment for this. Currently we're using LoginIds to save credentials to CredMan. However, with GH Enterprise server, since LoginId isn't unique enough, we're switching to URLs (we should just have been doing this always). When the new version of DevHome (with this PR) runs for the first time on a machine which already has saved credentials from a previous version, these credentials would have been saved with LoginIds (we switch these out for URLs on line 345-355). All logins on the this version will use URLs to save to CredMan. Will add a comment to make this part clear.